src/lcg.js

```diff
@@ -0,0 +1,7 @@
+export default seed => {
+  const a = 1664525,
+    c = 1013904223,
+    m = 4294967296; // 2^32
+  let s = Math.abs(a * +seed) || 1;
+  return () => (s = (a * s + c) % m) / m;
+};
```

Comment:
```suggestion
const a = 1664525;
const c = 1013904223;
const m = 4294967296; // 2^32

export default function lcg(s = 1) {
  if (!(s >= 0)) throw new Error(“invalid seed”);
  return () => (s = (a * s + c) % m) / m;
}
```

(I can’t figure out how to get iOS to type non-smart quotes…)

Reply:
I'm fine with the seed being used as given—for reference multiplying by *a* was motivated by this comment https://github.com/d3/d3-force/pull/160#commitcomment-40514779

Can you clarify why the default export should be a named function in this case?

Reply:
Another question here, again out of curiosity: when we define constants outside the function, don't they "exist" in memory even if they're not used? (Not that it matters here, but what if we needed to init a big chunk of memory?)

Reply:
Regarding the seed, it’s a question of what the expected seed values are. If the expected value of a seed is between 0 and *m* - 1, then we shouldn’t do anything with it. If the expected value is a float between 0 and 1, or any JavaScript number, then we should make suitable adjustments. But my take was to make the definition as simple as possible and expect the caller to provide a good seed — but we should document what a good seed is, and probably give examples, e.g., using a large number for the seed rather than 1, 2, or 3.

As for the named function, that’s totally optional. It’s nice for debugging the nonminified JavaScript sometimes but I think often it ends up getting stripped or renamed by Rollup. That name is generally only visible within the debugger (though it can also be used for recursive invocations, but not needed here).

Reply:
Thank you for the explanations.

I've now "fixed" the seed, so we accept any number: if negative it will be negated, and numbers < 1 will be inflated, so it matches usual expectations. seedrandom accepts string seeds, which is nice, but would need more code. If we wanted to offer seedrandom maybe we could share its seed generator. But I don't think we need to overcomplicate this and I'm happy with the current state of the PR.